src/scalar.rs

```diff
@@ -270,12 +270,33 @@ impl Scalar {
     /// require specific bit-patterns when performing scalar
     /// multiplication.
     pub const fn from_bits(bytes: [u8; 32]) -> Scalar {
-        let mut s = Scalar { bytes };
-        // Ensure that s < 2^255 by masking the high bit
-        s.bytes[31] &= 0b0111_1111;
+        let mut bytes:[u8; 32] = bytes;
+        bytes[31] &= 0b0111_1111;
+        Scalar { bytes }
```

Comment:
This will require making the above `bytes` param `mut`, i.e. `mut bytes: [u8; 32]`.

```suggestion
        bytes[31] &= 0b0111_1111;
        Scalar { bytes }
```

Reply:
Reverted
I actually checked this on godbolt. creating `mut Scalar` consumes less stack.